test/fx/test_verifier.py

```diff
@@ -0,0 +1,187 @@
+    is_valid_exir,
+)
+
+
+from typing import List, Optional, Tuple, Union
+
+
+from torch.testing._internal.common_utils import TestCase
+import torch  # noqa: F401
+import torch.nn as nn
+from torch import Tensor
+import torch._dynamo as torchdynamo
+import copy
+from functorch import make_fx
+from functorch.experimental import functionalize
+
+
+@torch.no_grad()
+def capture(f, args):
+    torchdynamo.config.capture_scalar_outputs = True
+    torchdynamo.config.guard_nn_modules = True
+    torchdynamo.config.dynamic_shapes = True
+    torchdynamo.config.specialize_int_float = True
+    torchdynamo.config.allow_rnn = True
+    torchdynamo.config.verbose = True
+    torchdynamo.reset()
+    f, _ = torchdynamo.export(
+                f,
+                *copy.deepcopy(args),
+                aten_graph=True,
+                tracing_mode='fake',
+            )
+    f = functionalize(f, remove='mutations')
+    gm = make_fx(f, tracing_mode='fake', _allow_non_fake_inputs=True)(*args)
+    return gm
+
+
+class Transpose(nn.Module):
+    def __init__(self) -> None:
+        super().__init__()
+
+    def forward(self, x: Tensor, dim0: int, dim1: int) -> Tensor:
+        return x.transpose(dim0, dim1)
+
+
+class Mul(nn.Module):
+    def __init__(self) -> None:
+        super().__init__()
+
+    def forward(self, input: Tensor, other: Tensor) -> Tensor:
+        # or return torch.mul(input, other)
+        return input * other
+
+    def get_random_inputs(self) -> Tuple[Tensor, Tensor]:
+        return (torch.randn(3, 2), torch.randn(3, 2))
+
+
+class ElementwiseAdd(nn.Module):
+    def __init__(self) -> None:
+        super().__init__()
+
+    def forward(self, x: Tensor, y: Tensor) -> Tensor:
+        return x + y
+
+    def get_random_inputs(self) -> Tuple[Tensor, Tensor]:
+        return (torch.randn(1, 3), torch.randn(1, 3))
+
+
+class Cat(nn.Module):
+    def __init__(self) -> None:
+        super().__init__()
+
+    # def forward(self, tensors, dim=0):
+    def forward(self, *args: Tensor, dim: int) -> Tensor:
+        tensors = args[:-1]
+        return torch.cat(tensors, dim)
+
+
+class FeedForwardBlock(nn.Module):
+    def __init__(self, input_dim: int, hidden_dim: int) -> None:
+        super().__init__()
+        self.input_dim = input_dim
+        self.hidden_dim = hidden_dim
+
+        self.layer_norm = nn.LayerNorm(input_dim)
+
+        self.relu = nn.ReLU()
+
+        self.linear1 = nn.Linear(input_dim, hidden_dim)
+        self.dropout1 = nn.Dropout()
+
+        self.linear2 = nn.Linear(hidden_dim, input_dim)
+        self.dropout2 = nn.Dropout()
+
+    def forward(self, x: Tensor) -> Tensor:
+        # LayerNorm -> Linear -> Dropout -> ReLU -> Linear -> Dropout
+        y = self.layer_norm(x)
+        y = self.linear1(y)
+        y = self.dropout1(y)
+        y = self.relu(y)
+        y = self.linear2(y)
+        y = self.dropout2(y)
+        return y
+
+
+
+class VerifierTest(TestCase):
+
+    def test_exir_verifier(self) -> None:
```

Comment:
do we need to mention `exir`? I feel like it's just an internal name, but when published we should have a different name.

Reply:
done. removed all mentions to exir.